sklearn/preprocessing/_data.py

```diff
@@ -2675,6 +2676,8 @@ def fit(self, X, y=None):
         self : object
            Fitted transformer.
         """
+        if self.subsample is None:
+            self.subsample = np.iinfo(int).max
```

Comment:
I should have been clearer: constructor arguments should be stored in a class attribute and not modified. Not even in `fit`. A guide for the peculiarities of how scikit-learn estimators are designed https://scikit-learn.org/stable/developers/develop.html#estimators

I also don't like using `np.iinfo(int).max` as a "magic value" here. We already have a special value (`None`) that says "turn off sub-sampling". Instead I would modify the `if` statements in `_dense_fit` and `_sparse_fit` to be `if self.subsample is not None and self.subsample < n_samples:` (not verbatim for sparse fit but same idea). You will also have to come up with a good modification for the check that the number of quantiles isn't bigger than `subsample` at the start of `fit`.

Reply:
Fair enough, I wanted to avoid having the same check (`is not None`) multiple times in a single execution path.